scripts/linux/runE2ETest.sh

```diff
@@ -0,0 +1,617 @@
+
+        print_highlighted_message "DIRECT MTEHOD RECEIVER LOGS"
+        sudo docker logs DirectMethodReceiver || true
+    fi
+}
+
+function process_args() {
+    print_highlighted_message "Process arguments"
+    saveNextArg=0
+    for arg in "$@"
+    do
+        if [ $saveNextArg -eq 1 ]; then
+            E2E_TEST_DIR="$arg"
+            saveNextArg=0
+        elif [ $saveNextArg -eq 2 ]; then
+            RELEASE_LABEL="$arg"
+            saveNextArg=0
+        elif [ $saveNextArg -eq 3 ]; then
+            ARTIFACT_IMAGE_BUILD_NUMBER="$arg"
+            saveNextArg=0
+        elif [ $saveNextArg -eq 4 ]; then
+            TEST_NAME="$arg"
+            saveNextArg=0
+        elif [ $saveNextArg -eq 5 ]; then
+            CONTAINER_REGISTRY="$arg"
+            saveNextArg=0
+        elif [ $saveNextArg -eq 6 ]; then
+            CONTAINER_REGISTRY_USERNAME="$arg"
+            saveNextArg=0
+        elif [ $saveNextArg -eq 7 ]; then
+            CONTAINER_REGISTRY_PASSWORD="$arg"
+            saveNextArg=0
+        elif [ $saveNextArg -eq 8 ]; then
+            IOTHUB_CONNECTION_STRING="$arg"
+            saveNextArg=0
+        elif [ $saveNextArg -eq 9 ]; then
+            EVENTHUB_CONNECTION_STRING="$arg"
+            saveNextArg=0
+        else
+            case "$arg" in
+                "-h" | "--help" ) usage;;
+                "-testDir" ) saveNextArg=1;;
+                "-releaseLabel" ) saveNextArg=2;;
+                "-artifactImageBuildNumber" ) saveNextArg=3;;
+                "-testName" ) saveNextArg=4;;
+                "-containerRegistry" ) saveNextArg=5;;
+                "-containerRegistryUsername" ) saveNextArg=6;;
+                "-containerRegistryPassword" ) saveNextArg=7;;
+                "-iotHubConnectionString" ) saveNextArg=8;;
+                "-eventHubConnectionString" ) saveNextArg=9;;
+                * ) usage;;
+            esac
+        fi
+    done
+
+    # Required parameters
+    [[ -z "$RELEASE_LABEL" ]] && { print_error "Release label is required."; exit 1; }
+    [[ -z "$ARTIFACT_IMAGE_BUILD_NUMBER" ]] && { print_error "Artifact image build number is required"; exit 1; }
+    [[ -z "$TEST_NAME" ]] && { print_error "Test name is required"; exit 1; }
+    [[ -z "$CONTAINER_REGISTRY_PASSWORD" ]] && { print_error "Container registry password is required"; exit 1; }
+    [[ -z "$IOTHUB_CONNECTION_STRING" ]] && { print_error "IoT hub connection string is required"; exit 1; }
+    [[ -z "$EVENTHUB_CONNECTION_STRING" ]] && { print_error "Event hub connection string is required"; exit 1; }
+
+    echo "Required parameters are provided"
+}
+
+function run_all_tests()
+{
+    local funcRet=0
+    local testRet=0
+
+    TEST_NAME="DirectMethodAmqp"
+    run_directmethodamqp_test && funcRet=$? || funcRet=$?
+
+    TEST_NAME="DirectMethodAmqpws"
+    run_directmethodamqpws_test && testRet=$? || testRet=$?
+    if (( $funcRet = 0 )); then funcRet=$testRet; fi
```

Comment:
You don't need `$` for variables inside `(())`. (shellcheck has an error for this. Please run it.)

Reply:
Ran shellcheck and resolved all issues.